### PR TITLE
refactor: Migrate VRF to equinix-sdk-go and move into separate package

### PR DESCRIPTION
**Pull request by aayushrangwala**
This updates the Equinix Metal project resource and data source from the deprecated packngo SDK to equinix-sdk-go and moves them into separate package required for supporting plugin framework

Relates to https://github.com/equinix/terraform-provider-equinix/issues/402

### CODE REVIEW

File: internal/resources/metal/vrf/datasource.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func dataSourceMetalVRF() *schema.Resource {
+func DataSource() *schema.Resource {
```
Comment (ctreatma) [inline internal/resources/metal/vrf/datasource.go:10]: While we're in here, let's add a test of the data source in `datasource_test.go` (I realize it shares almost all of its code with the resource, but that may not always be the case, so having a test helps protect against unintended consequences later).

File: equinix/resource_metal_virtual_circuit_acc_test.go [hunk 2]
```diff
@@ -14,6 +14,10 @@ import (
 	"github.com/packethost/packngo"
 )
 
+const (
```
Comment (ctreatma) [inline equinix/resource_metal_virtual_circuit_acc_test.go:17]: I think this might've leaked in from another change?

Comment (aayushrangwala) [reply equinix/resource_metal_virtual_circuit_acc_test.go:17]: No, this constant was being reused in virtual circuit as well. Hence had to copy that separately

File: internal/resources/metal/vrf/datasource_test.go [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+	var vrf metalv1.Vrf
+	rInt := acctest.RandInt()
+
+	datasourceName := "data.equinix_metal_vrf.foobar"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                  func() { acceptance.TestAccPreCheckMetal(t) },
+		PreventPostDestroyRefresh: true,
+		ExternalProviders:         acceptance.TestExternalProviders,
+		ProtoV5ProviderFactories:  acceptance.ProtoV5ProviderFactories,
```
Comment (ctreatma) [inline internal/resources/metal/vrf/datasource_test.go:89]: It looks like these lines should be removed for the upgrade tests.  We can only declare providers at 1 level; I looked at the upgrade test for the SSH key resource and it doesn't have these provider declarations, only the ones you already have under `Steps`.

```suggestion
```

Comment (aayushrangwala) [reply internal/resources/metal/vrf/datasource_test.go:89]: Yes right removed that and have on my local. Receiving different error now. Trying to fix that

Comment (ocobles) [reply internal/resources/metal/vrf/datasource_test.go]: the two `_upgradeFromVersion` should be removed as this resource has not yet been migrated to the framework and then you cannot use the ProtoV5ProviderFactories

**Pull request merged**